arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/Fold.kt

```diff
@@ -253,18 +252,32 @@ internal class DefaultRaise(@PublishedApi internal val isTraced: Boolean) : Rais
   @PublishedApi
   internal fun complete(): Boolean = isActive.getAndSet(false)
   override fun raise(r: Any?): Nothing = when {
-    isActive.value -> throw if (isTraced) RaiseCancellationException(r, this) else RaiseCancellationExceptionNoTrace(r, this)
+    isActive.value -> throw if (isTraced) Traced(r, this) else NoTrace(r, this)
     else -> throw RaiseLeakedException()
   }
 }
 
-/** CancellationException is required to cancel coroutines when raising from within them. */
-private class RaiseCancellationExceptionNoTrace(val raised: Any?, val raise: Raise<Any?>) :
-  CancellationExceptionNoTrace()
+@MustBeDocumented
+@Retention(AnnotationRetention.BINARY)
+@RequiresOptIn(RaiseCancellationExceptionCaptured, RequiresOptIn.Level.WARNING)
+public annotation class DelicateArrowApi
```

Comment:
```suggestion
public annotation class DelicateRaiseApi
```

It might be the case that we have several "delicate Arrow APIs" in the future, I think it's better to be explicit about this being the `Raise` one.

Reply:
Sounds good, I was in doubt between the two 😁 We have 1 instance of this already in Resource for `allocated`, but there I've re-used KotlinX's DelicateCoroutineApi which is already used in KotlinX for resources that need to be manually closed.